src/phpDocumentor/Reflection/Php/ProjectFactoryStrategyContainer.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * phpDocumentor
+ *
+ * PHP Version 5.5
+ *
+ * @copyright 2010-2015 Mike van Riel / Naenius (http://www.naenius.com)
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Reflection\Php;
+
+use phpDocumentor\Reflection\Exception;
+use phpDocumentor\Reflection\Php\ProjectFactoryStrategy;
+
+final class ProjectFactoryStrategyContainer implements StrategyContainer
+{
+    /**
+     * @var ProjectFactoryStrategy[]
+     */
+    private $strategies;
+
+    /**
+     * Initializes the factory with a number of strategies.
+     *
+     * @param ProjectFactoryStrategy[] $strategies
+     */
+    public function __construct($strategies)
+    {
+        foreach ($strategies as $strategy) {
+            if (!$strategy instanceof ProjectFactoryStrategy) {
```

Comment:
I think it would be nice to move this check to a public addStrategy method so that it could be re-used